src/services/fetchers/PolygonIOCurrencySnapshotGramsFetcher.ts

```diff
@@ -21,15 +25,36 @@ class PolygonIOCurrencySnapshotGramsFetcher extends BasePolygonIOSingleFetcher {
     this.valuePath = '$.ticker.lastQuote.a';
   }
 
-  async apply(params: {ticker: string}): Promise<number> {
+  async apply(params: {ticker: string}, options: FeedFetcherOptions): Promise<number> {
     const {ticker} = params;
+    const {base: feedBase, quote: feedQuote, timestamp} = options;
     const sourceUrl = `https://api.polygon.io/v2/snapshot/locale/global/markets/forex/tickers/${ticker}?apiKey=${this.apiKey}`;
 
+    if (!timestamp || timestamp <= 0) throw new Error(`${this.logPrefix} invalid timestamp value: ${timestamp}`);
+
     this.logger.debug(`${this.logPrefix} call for ${ticker}`);
 
     const data = await this.fetch(sourceUrl);
     const oneOzInGrams = 31.1034; // grams
-    return (data as number) / oneOzInGrams;
+    const price = (data as number) / oneOzInGrams;
```

Comment:
```suggestion
    const price = parseFloat(data) / oneOzInGrams;
```

Reply:
this is not possible right away without modifying extra code, since the function `fetch` returns this:

```typescript
async fetch(sourceUrl: string, raw = false): Promise<SinglePriceResponse | number>
```
and it is not possible to convert that to a string directly.